File: backend/src/infrastructure/database/models/study_port.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Float, Text, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from src.infrastructure.database.session import Base

class StudyPortModel(Base):
    __tablename__ = "study_ports"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    project_id = Column(UUID(as_uuid=True), ForeignKey("projects.id", ondelete="CASCADE"), nullable=False, unique=True)
    name = Column(String(255), nullable=False)
    code = Column(String(50), nullable=True)
    location = Column(String(255), nullable=False)
    latitude = Column(Float, nullable=False)
    longitude = Column(Float, nullable=False)
    description = Column(Text, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    # Relationships
    project = relationship("ProjectModel", back_populates="study_port")

File: frontend/src/types/studyPort.ts
export interface StudyPort {
  id: string;
  project_id: string;
  name: string;
  code?: string | null;
  location: string;
  latitude: number;
  longitude: number;
  description?: string | null;
  created_at: string;
  updated_at: string;
}

export interface StudyPortCreate {
  name: string;
  code?: string;
  location: string;
  latitude: number;
  longitude: number;
  description?: string;
}

export interface StudyPortUpdate {
  name?: string;
  code?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  description?: string;
}
